flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/metadata/FlinkRelMdFilteredColumnInterval.scala

```diff
@@ -132,11 +143,12 @@ class FlinkRelMdFilteredColumnInterval private extends MetadataHandler[FilteredC
     } else {
       val filterRef = calc.getProgram.getProjectList.get(filterArg)
       val condition = calc.getProgram.expandLocalRef(filterRef)
+      val columnRef = calc.getProgram.getProjectList.get(columnIndex).getIndex
```

Comment:
we need the similar logic with Project ? (should use the ref index of expandLocalRef result)

Reply:
good catch！